genesis/genesis_test.go

```diff
@@ -32,14 +36,173 @@ func TestAliases(t *testing.T) {
+				"rewardAddress": "X-local18jma8ppw3nhx5r4ap8clazz0dps7rv5u00z96u",
+				"delegationFee": 1000000
+			},
+			{
+				"nodeID": "NodeID-MFrZFVCXPv5iCn6M9K6XduxGTYp891xXZ",
+				"rewardAddress": "X-local18jma8ppw3nhx5r4ap8clazz0dps7rv5u00z96u",
+				"delegationFee": 500000
+			},
+			{
+				"nodeID": "NodeID-NFBbbJ4qCmNaCzeW7sxErhvWqvEQMnYcN",
+				"rewardAddress": "X-local18jma8ppw3nhx5r4ap8clazz0dps7rv5u00z96u",
+				"delegationFee": 250000
+			},
+			{
+				"nodeID": "NodeID-GWPcbFJZFfZreETSoWjPimr846mXEKCtu",
+				"rewardAddress": "X-local18jma8ppw3nhx5r4ap8clazz0dps7rv5u00z96u",
+				"delegationFee": 125000
+			},
+			{
+				"nodeID": "NodeID-P7oB2McjBGgW2NXXWVYjV8JEDFoW9xDE5",
+				"rewardAddress": "X-local18jma8ppw3nhx5r4ap8clazz0dps7rv5u00z96u",
+				"delegationFee": 62500
+			}
+		],
+		"cChainGenesis": "{\"config\":{\"chainId\":43112,\"homesteadBlock\":0,\"daoForkBlock\":0,\"daoForkSupport\":true,\"eip150Block\":0,\"eip150Hash\":\"0x2086799aeebeae135c246c65021c82b4e15a2c451340993aacfd2751886514f0\",\"eip155Block\":0,\"eip158Block\":0,\"byzantiumBlock\":0,\"constantinopleBlock\":0,\"petersburgBlock\":0,\"istanbulBlock\":0,\"muirGlacierBlock\":0},\"nonce\":\"0x0\",\"timestamp\":\"0x0\",\"extraData\":\"0x00\",\"gasLimit\":\"0x5f5e100\",\"difficulty\":\"0x0\",\"mixHash\":\"0x0000000000000000000000000000000000000000000000000000000000000000\",\"coinbase\":\"0x0000000000000000000000000000000000000000\",\"alloc\":{\"0100000000000000000000000000000000000000\":{\"code\":\"0x7300000000000000000000000000000000000000003014608060405260043610603d5760003560e01c80631e010439146042578063b6510bb314606e575b600080fd5b605c60048036036020811015605657600080fd5b503560b1565b60408051918252519081900360200190f35b818015607957600080fd5b5060af60048036036080811015608e57600080fd5b506001600160a01b03813516906020810135906040810135906060013560b6565b005b30cd90565b836001600160a01b031681836108fc8690811502906040516000604051808303818888878c8acf9550505050505015801560f4573d6000803e3d6000fd5b505050505056fea26469706673582212201eebce970fe3f5cb96bf8ac6ba5f5c133fc2908ae3dcd51082cfee8f583429d064736f6c634300060a0033\",\"balance\":\"0x0\"}},\"number\":\"0x0\",\"gasUsed\":\"0x0\",\"parentHash\":\"0x0000000000000000000000000000000000000000000000000000000000000000\"}",
+		"message": "{{ fun_quote }}"
+	}`
+	invalidGenesisConfigJSON = `{
+		"networkID": 9999}}}}
+	}`
+)
+
 func TestGenesis(t *testing.T) {
-	genesisBytes, _, err := Genesis(constants.MainnetID)
-	if err != nil {
-		t.Fatal(err)
+	tests := map[string]struct {
+		networkID       uint32
+		customConfig    string
+		missingFilePath string
+		err             string
+		expected        string
+	}{
+		"mainnet": {
+			networkID: constants.MainnetID,
+			expected:  "3e6662fdbd88bcf4c7dd82cb4699c0807f1d7315d493bc38532697e11b226276",
+		},
+		"fuji": {
+			networkID:    constants.FujiID,
+			customConfig: localGenesisConfigJSON, // won't load
+			expected:     "2e6b699298a664793bff42dae9c1af8d9c54645d8b376fd331e0b67475578e0a",
```

Comment:
there is probably a better way to do this....

Reply:
I thought about using an anonymous function to load these on-the-fly but didn't like the look.